src/bin/server/worker.rs

```diff
@@ -44,6 +46,12 @@ impl Worker {
             index_file,
             extract_config: Default::default(),
             github: rla::github::Client::new()?,
+            // We have ~70 builders so let's keep around 100 entries, or around
+            // 1 PRs worth of tests. This allows up to ~30 interleaved items
+            // received by this worker (from PR builds and such). If we're
+            // seeing walls of replies about the same thing failing, this should
+            // probably be bumped up.
+            seen: RecentSeen::new(100),
```

Comment:
Can you move the length in a constant at the top so it's easier to update? If you go with the ring buffer route a comment saying not to increase that number a lot due to the O(n) issue would be nice.